### PR TITLE
initial flags modification PR. Open to review.

### PR DESCRIPTION
**Pull request by DSMishler**
I understand this is unlikely to be immediately approved, but just wanted to put it on a PR officially before our weekly meeting.

TLDR: add a flag so that task timing can be done more accurately. The results? It turns out it does make a little bit of a difference.

### CODE REVIEW

File: parsec/profiling.h [hunk 1]
```diff
@@ -85,26 +85,34 @@
...
-#define PARSEC_PROFILING_EVENT_COUNTER      (1<<2)
+#define PARSEC_PROFILING_EVENT_COUNTER          (1<<2)
 /**
  * @brief Constant to use when no handle/object ID can be associated
  *        with an event
  * @details Some functions take a handle/object ID as a parameter.
  *          When such ID is not applicable, this constant should be
  *          passed to the parameter.
  */
+#define PARSEC_PROFILING_EVENT_TIME_AT_START    (1<<3)
```
Comment (therault) [inline parsec/profiling.h:108]: The comment goes before the constant definition. Doxygen will match the wrong comment.

Comment (therault) [inline parsec/profiling.h:108]: You should also define a PARSEC_PROFILING_EVENT_TIME_AT_END as 0 and use it symmetrically in the pins profiler that you modify, just for the clarity of reading.

File: parsec/mca/pins/task_profiler/pins_task_profiler_module.c [hunk 2]
```diff
@@ -252,9 +252,10 @@ task_profiler_exec_count_end(struct parsec_execution_stream_s*   es,
 {
     if (NULL != task->taskpool->profiling_array &&
         task->task_class->task_class_id < task->taskpool->nb_task_classes)
-        PARSEC_TASK_PROF_TRACE(es->es_profile,
-                               task->taskpool->profiling_array[2*task->task_class->task_class_id+1],
-                               task);
+        PARSEC_TASK_PROF_TRACE_FLAGS(es->es_profile,
```
Comment (therault) [inline parsec/mca/pins/task_profiler/pins_task_profiler_module.c:255]: This reindentation is incorrect and should not be part of the PR

File: parsec/parsec_internal.h [hunk 3]
```diff
@@ -570,9 +570,19 @@ extern int device_delegate_begin, device_delegate_end;
...
+                           (TASK)->task_class->key_functions->          \
+                           key_hash((TASK)->task_class->make_key(       \
+                              (TASK)->taskpool, (TASK)->locals ), NULL), \
+                           (TASK)->taskpool->taskpool_id,               \
+                           (void*)&(TASK)->prof_info,                   \
+                           (FLAGS)); 
 #else
 #define PARSEC_TASK_PROF_TRACE(PROFILE, KEY, TASK)
 #define PARSEC_TASK_PROF_TRACE_IF(COND, PROFILE, KEY, TASK)
+#define PARSEC_TASK_PROF_TRACE_FLAGS(PROFILE, KEY, TASK)
```
Comment (therault) [inline parsec/parsec_internal.h:585]: Does not follow the prototype of the other case (#if defined(PARSEC_PROF_TRACE))

**Review by therault — CHANGES_REQUESTED** [review]
See comments for list of changes

File: parsec/mca/pins/task_profiler/pins_task_profiler_module.c (hunk 2 above)
Comment (DSMishler) [reply parsec/mca/pins/task_profiler/pins_task_profiler_module.c:255]: I'm not sure if you meant re-indenting for `task` or reindenting everything according to the beginning of the arguments. My followup will assume you meant the latter.

**Review by therault — COMMENTED** [review]
Changes look good, but the CI checks are failing because of a Spack update.

Omri Mor suggests to try removing `concretization: together` and adding
```
concretizer:
   unify: true
```

to the file [.github/CI/github_runner.yaml](https://github.com/ICLDisco/parsec/tree/master/.github/CI/github_runner.yaml)

Can you try this?

**Comment by DSMishler** [comment]
Should be all set to go now!
